tyr/clusters/nginx.py

```diff
@@ -0,0 +1,113 @@
+import logging
+from tyr.servers.nginx import NginxServer
+from tyr.clusters.autoscaling import AutoScaler
+
+
+class NginxCluster():
+
+    def __init__(self, group=None,
+                 server_type=None,
+                 instance_type=None,
+                 environment=None,
+                 ami=None,
+                 region=None,
+                 tags=None,
+                 root_volume_size=8,
+                 subnet_ids=[],
+                 role=None,
+                 keypair=None,
+                 security_groups=None,
+                 autoscaling_group=None,
+                 desired_capacity=1,
+                 max_size=1,
+                 min_size=1,
+                 default_cooldown=300,
+                 availability_zones=None,
+                 health_check_grace_period=300,
+                 launch_configuration=None):
+
+        self.log = logging.getLogger('Tyr.Clusters.NginxCluster')
+        self.group = group
+        self.server_type = server_type
+        self.instance_type = instance_type
+        self.environment = environment
+        self.ami = ami
+        self.region = region
+        self.role = role
+        self.tags = tags
+        self.root_volume_size = root_volume_size
+        self.subnet_ids = subnet_ids
+
+        self.keypair = keypair
+        self.security_groups = security_groups
+        self.autoscaling_group = autoscaling_group
+        self.desired_capacity = desired_capacity
+        self.max_size = max_size
+        self.min_size = min_size
+        self.default_cooldown = default_cooldown
+        self.availability_zones = availability_zones
+        self.health_check_grace_period = health_check_grace_period
+        self.launch_configuration = launch_configuration
+
+        if 'Role' not in self.tags:
+          self.tags.append({'name':'Role', 'value':'RoleNginx'})
```

Comment:
Role should always be added to tags even if tags = None (tags always at least contains 'Role' of 'RoleNginx').